pvlib/iotools/psm3.py

```diff
@@ -133,6 +157,13 @@ def get_psm3(latitude, longitude, api_key, email, names='tmy', interval=60,
     # convert to string to accomodate integer years being passed in
     names = str(names)
 
+    # convert pvlib names in attributes to psm3 convention (reverse mapping)
+    # unlike psm3 columns, attributes are lower case and with underscores
+    amap = {value: key.lower().replace(' ', '_') for (key, value) in
+            PSM3_VARIABLE_MAP.items()}
+    attributes = [a if a not in amap.keys() else amap[a] for a in attributes]
+    attributes = list(set(attributes))  # remove duplicate values
```

Comment:
This is interesting.  Do we do "reverse mapping" in any other iotools functions?  It is unfortunate that the PSM3 API's input parameter names are different from the output column names.

Maybe clearer for the second line, up to you: `attributes = [amap.get(a, a) for a in attributes]`

Reply:
If we keep this, it should probably be mentioned in the docstring

Reply:
I cannot immediately think of any other function where reverse mapping would make sense. Pretty nifty though! And I think it is in line with the spirit of the iotools, in that it conforms the data access interface with pvlib conventions.

Reply:
@AdamRJensen should we add a sentence to the `attributes` docstring description for this?  Something like (feel free to edit):
> Alternatively, pvlib names may also be used (e.g. ``'ghi'`` rather than ``'GHI'``); see :const:`PSM3_VARIABLE_MAP`.

Reply:
If we're referencing the iotools variable maps via `:const:` then we should think about giving them their own entries in the docs.  Something like what we did for `pvlib.temperature.TEMPERATURE_MODEL_PARAMETERS` maybe?  Let's do that in a separate issue though.